nat-handler.py

```diff
@@ -94,17 +102,30 @@ def current_az(self):
     def current_region(self):
         return self.current_az[:-1]
 
+    @property
+    def primary_interface_id(self):
+        ec2 = connect_to_ec2(self.current_region)
+        instance = ec2.get_only_instances(instance_ids=[self.current_instance_id])[0]
+        interfaces = instance.interfaces
+        interface_id = [i.id for i in interfaces if i.id != self.secondary_interface_id]
+        return interface_id[0]
+
+    @property
+    def secondary_interface_id(self):
+        return self.config.secondary_interface_id(self.current_az)
+
     def take_route(self, az):
         route_table_id = self.config.route_table_id(az)
         vpc = connect_to_vpc(self.current_region)
-        vpc.replace_route(route_table_id, '0.0.0.0/0', instance_id=self.current_instance_id)
+        vpc.replace_route(route_table_id, '0.0.0.0/0', interface_id=self.primary_interface_id)
```

Comment:
No. It won't work. We need to have elastic IPs. I'm not sure how any of this would work without elastic IPs as the failover is to use EIPs to be associated with ENIs (or the instance for that matter). So we need elastic IPs anyway right?

Do you have any examples where elastic IPs won't be needed?

I am also not aware of the development/ staging setup for this and the use of EIPs for accepting auths.